internal/services/managedredis/managed_redis_resource.go

```diff
@@ -0,0 +1,755 @@
+//
+// There might be a plan to support multiple databases in the future, in which case we will implement an
+// azurerm_managed_redis_custom_db resource.
+
+type ManagedRedisResource struct{}
+
+var _ sdk.ResourceWithUpdate = ManagedRedisResource{}
+
+type ManagedRedisResourceModel struct {
+	Name              string `tfschema:"name"`
+	ResourceGroupName string `tfschema:"resource_group_name"`
+
+	Location string `tfschema:"location"`
+
+	SkuName string `tfschema:"sku_name"`
+
+	CustomerManagedKey      []CustomerManagedKeyModel                  `tfschema:"customer_managed_key"`
+	DefaultDatabase         []DefaultDatabaseModel                     `tfschema:"default_database"`
+	HighAvailabilityEnabled bool                                       `tfschema:"high_availability_enabled"`
+	Identity                []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+	Tags                    map[string]string                          `tfschema:"tags"`
+
+	Hostname string `tfschema:"hostname"`
+}
+
+type CustomerManagedKeyModel struct {
+	KeyVaultKeyId          string `tfschema:"key_vault_key_id"`
+	UserAssignedIdentityId string `tfschema:"user_assigned_identity_id"`
+}
+
+type DefaultDatabaseModel struct {
+	AccessKeysAuthenticationEnabled bool          `tfschema:"access_keys_authentication_enabled"`
+	ClientProtocol                  string        `tfschema:"client_protocol"`
+	ClusteringPolicy                string        `tfschema:"clustering_policy"`
+	EvictionPolicy                  string        `tfschema:"eviction_policy"`
+	GeoReplicationGroupName         string        `tfschema:"geo_replication_group_name"`
+	Module                          []ModuleModel `tfschema:"module"`
+
+	Port               int64  `tfschema:"port"`
+	PrimaryAccessKey   string `tfschema:"primary_access_key"`
+	SecondaryAccessKey string `tfschema:"secondary_access_key"`
+}
+
+type ModuleModel struct {
+	Name    string `tfschema:"name"`
+	Args    string `tfschema:"args"`
+	Version string `tfschema:"version"`
+}
+
+const defaultDatabaseName = "default"
+
+func (r ManagedRedisResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedRedisClusterName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"sku_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringInSlice(validate.PossibleValuesForSkuName(), false),
+		},
+
+		"customer_managed_key": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"key_vault_key_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.IsURLWithHTTPorHTTPS,
```

Comment:
Sorry, missed this, can we use the KV validator?
```suggestion
						ValidateFunc: keyVaultValidate.NestedItemId,
```
import as: 
```go
	keyVaultValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/validate"
```

Reply:
Pushed 78ffa34313, thx